### PR TITLE
fix(ios): amend Ti.UI.Shortcut implementation

### PR DESCRIPTION
**Pull request by jquick-axway**
**JIRA:**
https://jira.appcelerator.org/browse/TIMOB-26818

**Summary:**
- Modified `Ti.UI.Shortcut` to be a module.
- Removed `Ti.UI.createShortcut()` API.
- Temporarily disabled Android shortcut unit tests until it's been changed to match.

**Test:**
1. Build and run the below on iOS.
2. Tap on "Add Shortcut" button.
3. Tap on "Shortcut Exists?" button.
4. Verify alert shows shortcut info.
5. Tap on "List Added Shortcuts" button.
6. Verify an alert lists 1 shortcut.
7. Close the alert dialog.
8. Tap on the home button.
9. Tap and hold the app icon.
10. Verify popup shows "Shortcut Title" option.
11. Tap on the "Shortcut Title" menu item.
12. Verify app is shown in foreground with alert stating shortcut was clicked on.
13. Tap on "Remove 1 Shortcut" button.
14. Tap on "Shortcut Exists?" button.
15. Verify alert states shortcut was **NOT** found.
16. Tap on "List Added Shortcuts" button.
17. Verify alert does not list any shortcuts.
18. Tap on Home button.
19. Tap and hold the app icon.
20. Verify "Shortcut Title" no longer appears in popup.

```javascript
Ti.UI.Shortcut.addEventListener("click", function(e) {
	alert(
		"Clicked on shortcut:\n" +
		"- id: " + e.item.id + "\n" +
		"- title: " + e.item.title + "\n" +
		"- description: " + e.item.description);
});

var shortcutItem = Ti.UI.createShortcutItem({
	id: "shortcut_item_1",
	title: "Shortcut Title",
	description: "Shortcut Description",
	icon: Ti.UI.iOS ? Ti.UI.iOS.SHORTCUT_ICON_TYPE_ADD : undefined,
});

var window = Ti.UI.createWindow({ layout: "vertical" });
var addButton = Ti.UI.createButton({
	title: "Add Shortcut",
	top: Ti.App.iOS ? "30dp" : "10dp",
});
addButton.addEventListener("click", function() {
	Ti.UI.Shortcut.add(shortcutItem);
});
window.add(addButton);
var existsButton = Ti.UI.createButton({
	title: "Shortcut Exist?",
	top: "10dp",
});
existsButton.addEventListener("click", function() {
	var item = Ti.UI.Shortcut.getById(shortcutItem.id);
	if (item) {
		alert(
			"Shortcut '" + item.id + "' was found.\n" +
			"- Title: " + item.title + "\n" +
			"- Description: " + item.description);
	} else {
		alert("Shortcut '" + shortcutItem.id + "' NOT found.");
	}
});
window.add(existsButton);
var removeByReferenceButton = Ti.UI.createButton({
	title: "Remove 1 Shortcut",
	top: "10dp",
});
removeByReferenceButton.addEventListener("click", function() {
	Ti.UI.Shortcut.remove(shortcutItem);
});
window.add(removeByReferenceButton);
var removeAllButton = Ti.UI.createButton({
	title: "Remove All Shortcuts",
	top: "10dp",
});
removeAllButton.addEventListener("click", function() {
	Ti.UI.Shortcut.removeAll();
});
window.add(removeAllButton);
var getAllButton = Ti.UI.createButton({
	title: "List Added Shortcuts",
	top: "10dp",
});
getAllButton.addEventListener("click", function() {
	let message = "Ti.UI.Shortcut.items:\n";
	for (const nextItem of Ti.UI.Shortcut.items) {
		message += "- " + nextItem.id + "\n";
	}
	alert(message);
});
window.add(getAllButton);
window.open();
```

### CODE REVIEW

File: iphone/Classes/TiUIShortcutProxy.h [hunk 1]
```diff
@@ -7,26 +7,33 @@
...
 READONLY_PROPERTY(NSArray<TiUIShortcutItemProxy *> *, items, Items);
 READONLY_PROPERTY(NSArray<TiUIShortcutItemProxy *> *, staticItems, StaticItems);
 
 - (TiUIShortcutItemProxy *)getById:(NSString *)identifier;
-
 - (void)remove:(TiUIShortcutItemProxy *)shortcut;
-
 - (void)removeAll;
 - (void)add:(TiUIShortcutItemProxy *)shortcut;
+*/
```
Comment (jquick-axway) [inline iphone/Classes/TiUIShortcutProxy.h:28]: I need to remove this comment block.

File: iphone/Classes/TiUIShortcutProxy.h [hunk 2]
```diff
@@ -7,26 +7,33 @@
...
 
 @class TiUIShortcutItemProxy;
 
+@interface TiUIShortcutProxy : TiProxy {
+  @private
+}
+
 NS_ASSUME_NONNULL_BEGIN
 
-@protocol TiUIShortcutProxyExports <JSExport>
```
Comment (vijaysingh-axway) [inline iphone/Classes/TiUIShortcutProxy.h:16]: Why this removed?

File: iphone/Classes/UIModule.m [hunk 3]
```diff
@@ -739,6 +745,16 @@ - (NSString *)CLIPBOARD_OPTION_EXPIRATION_DATE
 MAKE_SYSTEM_PROP(TABLE_VIEW_SEPARATOR_STYLE_NONE, UITableViewCellSeparatorStyleNone);
 MAKE_SYSTEM_PROP(TABLE_VIEW_SEPARATOR_STYLE_SINGLE_LINE, UITableViewCellSeparatorStyleSingleLine);
 
+#if defined(USE_TI_UISHORTCUT) || defined(USE_TI_UISHORTCUTITEM)
+- (TiUIShortcutProxy *)Shortcut
+{
+  if (shortcut == nil) {
+    shortcut = [[TiUIShortcutProxy alloc] _initWithPageContext:[self executionContext]];
```
Comment (vijaysingh-axway) [inline iphone/Classes/UIModule.m:752]:
```suggestion
    shortcut = [[TiUIShortcutProxy alloc] init];
```
And restoring the implementation in TiUIShortcutProxy should work. No need to change ObjcProxy style to old one.

File: iphone/Classes/UIModule.h [hunk 4]
```diff
@@ -29,7 +33,10 @@
 #ifdef USE_TI_UICLIPBOARD
   TiProxy *clipboard;
 #endif
+#if defined(USE_TI_UISHORTCUT) || defined(USE_TI_UISHORTCUTITEM)
```
Comment (vijaysingh-axway) [inline iphone/Classes/UIModule.h:36]: This should guard next line -
` TiUIShortcutProxy *shortcut;`

**Review by vijaysingh-axway — APPROVED** [review]
CR passed.

**Review by ssjsamir — COMMENTED** [review]
@jquick-axway When using the test case in the description and adding shortcuts to the app returns 0 'NUMBER OF SHORTCUTS' when clicking on the button even though shortcuts have been created.

**Test Environment**
{code:java}
MacOS Big Sur: 11.0 Beta
Xcode: 12.0 Beta 
Java Version: 1.8.0_242
Android NDK: 21.3.6528147
Node.js: 12.18.1
""NPM":"5.0.0","CLI":"8.1.0-master.5""
iphone 8 Sim (14.0 Beta)
iphone 10 (14.0 Beta)
{code}

**Comment by jquick-axway** [comment]
@ssjsamir, the issue is that iOS only supports showing 1 alert at a time and the test code is trying to show 2 alerts. The "static" shortcut should be zero in this case. The 2nd alert (which isn't appearing) shows the "dyncamic" shortcut count which will be 1. I'll update the test script. Sorry about the confusion.

**Review by ssjsamir — APPROVED** [review]
`FR Passed` tested using the new updated tests steps in the description.
**Test Environment**
```
MacOS Big Sur: 11.0 Beta
Xcode: 12.0 Beta
Java Version: 1.8.0_242
Android NDK: 21.3.6528147
Node.js: 12.18.1
""NPM":"5.0.0","CLI":"8.1.0-master.5""
iphone 8 Sim (14.0 Beta)
iphone 10 (14.0 Beta)
```

**Pull request merged**